mozci/data/contract.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+
+from dataclasses import dataclass
+from typing import Tuple
+
+from voluptuous import Any, Marker, Optional, Required, Schema
+
+
+@dataclass
+class Contract:
+    name: str
+    description: str
+    validate_in: Schema
+    validate_out: Schema
+
+
+_contracts: Tuple[Contract, ...] = (
+    Contract(
+        name="push_tasks",
+        description="Data about the tasks that ran on a given push.",
+        validate_in=Schema({Required("branch"): str, Required("rev"): str,}),
+        validate_out=Schema(
```

Comment:
This looks like a straightforward contract to comply with.  Not a problem in Treeherder.